admin_manual/configuration/occ_command.rst

```diff
@@ -0,0 +1,252 @@
+singleuser, and repair.
+
+You must put your ownCloud server into maintenance mode whenever you perform an 
+update or upgrade. This locks the sessions of all logged-in users, including 
+administrators, and displays a status screen warning that the server is in 
+maintenance mode. Users who are not already logged in cannot log in until 
+maintenance mode is turned off. When you take the server out of maintenance mode 
+logged-in users must refresh their Web browsers to continue working::
+
+ $ sudo -u www-data php occ maintenance:mode --on
+ $ sudo -u www-data php occ maintenance:mode --off
+ 
+Putting your ownCloud server into single-user mode allows admins to log in and 
+work, but not ordinary users. This is useful for performing maintenance and 
+troubleshooting on a running server::
+
+ $ sudo -u www-data php occ maintenance:singleuser --on
+   Single user mode enabled
+   
+And turn it off when you're finished::
+
+ $ sudo -u www-data php occ maintenance:singleuser --off
+   Single user mode disabled
+
+TODO: What does  maintenance:repair do? Needs details::
+  
+  $ sudo -u www-data php occ maintenance:repair
+    - Repair mime types  
+    - Repair config
+ 
+User Commands
+-------------
+
+The ``user`` commands reset passwords, display a simple report showing how 
+many users you have, and when a user was last logged in.
+
+You can reset any user's password, including administrators (see 
+:doc:`reset_admin_password`). In this example the username is layla::
+
+ $ sudo -u www-data php occ user:resetpassword layla
+   Enter a new password: 
+   Confirm the new password: 
+   Successfully reset password for layla
+   
+View a user's most recent login::   
+   
+ $ sudo -u www-data php occ user:lastseen layla 
+   layla`s last login: 09.01.2015 18:46
+   
+TODO: does this count LDAP and other external users, or local only?::
+
+ $ sudo -u www-data php occ user:report
+   +------------------+---+
+   | User Report      |   |
+   +------------------+---+
+   | OC_User_Database | 2 |
+   |                  |   |
+   | total users      | 2 |
+   |                  |   |
+   | user directories | 3 |
+   +------------------+---+
+   
+Apps Commands
+-------------
+
+The ``app`` commands list, enable, and disable apps. This lists all of your 
+installed apps, and shows whether they are enabled or disabled::
+
+ $ sudo -u www-data php occ app:list
+ 
+Enable an app::
+
+ $ sudo -u www-data php occ app:enable external
+   external enabled
+   
+Disable an app::
+
+ $ sudo -u www-data php occ app:disable external
+   external disabled
+   
+Upgrade Command
+---------------
+
+When you are performing an update or upgrade on your ownCloud server, it is 
+better to use ``occ`` to perform the database upgrade step, rather than the Web 
+GUI,  in order to avoid timeouts.
+
+TODO: what timeouts? What causes them?
+
+You can perform a dry-run first to see what will happen, without changing 
```

Comment:
A dry run will test if the database schema migration simulation will work – that is needed to verify if the update is possible from a DB PoV or if things will explode. I think this sentence needs clarifying.